segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessorTest.java

```diff
@@ -1724,6 +1728,85 @@ public void testTruncateIndexSegmentWithGreaterThenEndOffset() throws Exception
+                .startOffset(0)
+                .attributes(Map.of(EVENT_COUNT, 30L))
+                .build();
+
+        doThrow(new CompletionException(new StreamSegmentNotExistsException("Segment does not exits")))
+                .when(store).getStreamSegmentInfo(indexSegmentName, timeout);
+        doReturn(CompletableFuture.completedFuture(null)).when(store).truncateStreamSegment(anyString(), anyLong(), any());
+
+        Reply reply = sendRequest(channel, new WireCommands.TruncateSegment(requestId, segmentName, 30L, ""));
+        assertTrue(reply instanceof WireCommands.SegmentTruncated);
+    }
+
+    @Test
+    public void testTruncateIndexSegmentThrowIllegalArgumentException() throws Exception {
+        String segmentName = "truncateSegment";
+        String indexSegmentName = getIndexSegmentName(segmentName);
+        StreamSegmentStore store = mock(StreamSegmentStore.class);
+        @Cleanup
+        EmbeddedChannel channel = createChannel(store);
+
+        Duration timeout = Duration.ofMinutes(1);
+        ServiceBuilder serviceBuilder = newInlineExecutionInMemoryBuilder(getBuilderConfig());
+        serviceBuilder.initialize();
+
+        StreamSegmentInformation.builder()
+                .name(indexSegmentName)
+                .length(1000)
+                .startOffset(100)
+                .attributes(Map.of(EVENT_COUNT, 30L))
+                .build();
+
+        doThrow(new CompletionException(new IllegalArgumentException("Illegal argument-provided")))
+                .when(store).getStreamSegmentInfo(indexSegmentName, timeout);
+        doReturn(CompletableFuture.completedFuture(null)).when(store).truncateStreamSegment(anyString(), anyLong(), any());
+
+        Reply reply = sendRequest(channel, new WireCommands.TruncateSegment(requestId, segmentName, 30L, ""));
+        assertTrue(reply instanceof WireCommands.ErrorMessage);
+    }
+
+    @Test
+    public void testTruncateIndexSegmentThrowIllegalStateException() throws Exception {
+        String segmentName = "truncateSegment";
+        String indexSegmentName = getIndexSegmentName(segmentName);
+        StreamSegmentStore store = mock(StreamSegmentStore.class);
+        @Cleanup
+        EmbeddedChannel channel = createChannel(store);
+
+        Duration timeout = Duration.ofMinutes(1);
+        ServiceBuilder serviceBuilder = newInlineExecutionInMemoryBuilder(getBuilderConfig());
+        serviceBuilder.initialize();
+
+        StreamSegmentInformation.builder()
+                .name(indexSegmentName)
+                .length(1000)
+                .startOffset(100)
+                .attributes(Map.of(EVENT_COUNT, 30L))
+                .build();
+
+        doThrow(new CompletionException(new IllegalStateException("Illegal State")))
+                .when(store).getStreamSegmentInfo(indexSegmentName, timeout);
+        doReturn(CompletableFuture.completedFuture(null)).when(store).truncateStreamSegment(anyString(), anyLong(), any());
```

Comment:
Looks like this call will fail while truncating main segment itself. It will not reach to index segment truncation part.

Reply:
As said above.